File: trunk/OscarDotNet/component/biz/Class_biz_leave.pas
unit Class_biz_leave;

interface

type
  //
  filter_type =
    (
    NONE,
    BOTH,
      OBLIGATED,
      ON_LEAVE
    );
  //
  TClass_biz_leave = class
  private
    { Private Declarations }
  public
    constructor Create;
    function BeLeaf(filter: filter_type): boolean;
  end;

implementation

constructor TClass_biz_leave.Create;
begin
  inherited Create;
  // TODO: Add any constructor code here
end;

function TClass_biz_leave.BeLeaf(filter: filter_type): boolean;
begin
  BeLeaf := FALSE;
  if filter in [OBLIGATED,ON_LEAVE] then begin
    BeLeaf := TRUE;
  end;
end;

end.

File: component/biz/Class_biz_leave.cs
using System;
using System.Collections;
namespace Class_biz_leave
{
    public class TClass_biz_leave
    {
        //Constructor  Create()
        public TClass_biz_leave() : base()
        {
            // TODO: Add any constructor code here

        }
        public bool BeLeaf(filter_type filter)
        {
            bool result;
            result = false;
            if (new ArrayList(new filter_type[] {filter_type.OBLIGATED, filter_type.ON_LEAVE}).Contains(filter))
            {
                result = true;
            }
            return result;
        }

    } // end TClass_biz_leave

    public enum filter_type
    {
        NONE,
        BOTH,
        OBLIGATED,
        ON_LEAVE,
    } // end filter_type

}
